vespalib/src/vespa/vespalib/util/spin_lock.h

```diff
@@ -0,0 +1,33 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <atomic>
+
+namespace vespalib {
+
+/**
+ * A spin-lock implementation that favors uncontended performance.
+ * Some measures are taken to reduce the impact of threads waiting to
+ * get the lock since this will not affect the fast-path of obtaining
+ * the lock immediately.
+ *
+ * This implementation satisfies the BasicLockable requirements,
+ * making it work with things like std::lock_guard.
```

Comment:
Note that the spin lock does not provide priority inheritance, a low priority thread preempted while holding the spin lock can cause excessive spinning in a high priority thread if the system has few (e.g. a single) cpu cores.

Reply:
does std::mutex provide this, even if std::thread does not provide priority?

Reply:
added more warnings to the comment